### PR TITLE
RDRP-989 move geo cols to config

### PR DESCRIPTION
**Pull request by AnneONS**
## Pull Request submission

*Insert detailed bullet points about your changes here!*

*Insert any instructions to help the reviewer, e.g. "install new requirements from `requirements.txt`"*

*Let the reviewer know what data files are needed (and if applicable, where they are to be found)

#### Closes or fixes

* Detail the ticket(s) you are closing with this PR
Closes #

#### Code

- [ ] **Code runs** The code runs on my machine and/or CDSW
- [ ] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [ ] **Requirements** My/our code functions according to the requirements of the ticket
- [ ] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [ ] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [ ] **Clean Code**
    - [ ] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [ ] Code passess flake8 linting check
    - [ ] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [ ] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [ ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `requirements.txt`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

File: src/site_apportionment/site_apportionment.py [hunk 1]
```diff
@@ -217,13 +218,15 @@ def create_category_df(
         groupby_cols (List[str]): List of columns to group by.
         code_cols (List[str]): List of code columns.
         site_cols (List[str]): List of site column.
+        geo_cols (List[str]): List of geographic columns from the config.
```
Comment (JenCheshire) [inline src/site_apportionment/site_apportionment.py:221]: geo cols doesn't include itl - will that need adding so it appears in GBSAS correctly?  Not sure if that needs to be kept separate so it doesn't impact the intram itl output

File: src/site_apportionment/site_apportionment.py [hunk 2]
```diff
@@ -309,11 +313,8 @@ def create_sites_df(
     # De-duplicate by summing percents
     sites_df[percent_col] = sites_df[percent_col].fillna(0)
     agg_dict = {instance_col: "first", percent_col: "sum"}
-    sites_df = (
-        sites_df.groupby(groupby_cols + [postcode_col, postcodes_harmonised_col])
-        .agg(agg_dict)
-        .reset_index()
-    )
+    cols = [c for c in (site_cols + geo_cols) if c not in groupby_cols]
+    sites_df = sites_df.groupby(groupby_cols + cols).agg(agg_dict).reset_index()
```
Comment (JenCheshire) [inline src/site_apportionment/site_apportionment.py]: I'm getting a value error here: ValueError: cannot insert 602, already exists

File: src/site_apportionment/site_apportionment.py (hunk 1 above)
Comment (AnneONS) [reply src/site_apportionment/site_apportionment.py:221]: good point, thanks

File: src/site_apportionment/site_apportionment.py (hunk 2 above)
Comment (AnneONS) [reply src/site_apportionment/site_apportionment.py]: fixed a bug here, thanks

File: src/site_apportionment/site_apportionment.py [hunk 3]
```diff
@@ -294,11 +297,13 @@ def create_sites_df(
         df (pd.DataFrame): The input DataFrame.
         groupby_cols (List[str]): Columns to group by: reference, period.
         site_cols (List[str]): Columns of sites (instance, postcode, percent).
+        geo_cols
```
Comment (JenCheshire) [inline src/site_apportionment/site_apportionment.py]: Missing the detail

File: src/imputation/imputation_helpers.py [hunk 4]
```diff
@@ -272,7 +272,6 @@ def split_df_on_trim(df: pd.DataFrame, trim_bool_col: str) -> pd.DataFrame:
 
 
 def split_df_on_imp_class(df: pd.DataFrame, exclusion_list: List = ["817", "nan"]):
```
Comment (JenCheshire) [inline src/imputation/imputation_helpers.py:274]: No docstring

File: src/imputation/imputation_helpers.py [hunk 5]
```diff
@@ -353,13 +352,11 @@ def tidy_imputation_dataframe(
     to_impute_cols: List,
 ) -> pd.DataFrame:
     """Remove rows and columns not needed after imputation."""
```
Comment (JenCheshire) [inline src/imputation/imputation_helpers.py]: No args/returns

File: src/mapping/mapping_main.py [hunk 6]
```diff
@@ -25,14 +25,9 @@ def run_mapping(
     rd_file_exists: Callable,
     run_id: int,
 ):
-
```
Comment (JenCheshire) [inline src/mapping/mapping_main.py:28]: Docstring

File: src/outputs/short_form.py [hunk 7]
```diff
@@ -92,7 +92,6 @@ def output_short_form(
     config: Dict[str, Any],
     write_csv: Callable,
     run_id: int,
-    postcode_itl_mapper: pd.DataFrame,
 ):
     """Run the outputs module.
```
Comment (JenCheshire) [inline src/outputs/short_form.py:97]: ultfoc mapper can be taken out the arg's

File: src/outputs/tau.py [hunk 8]
```diff
@@ -26,7 +25,6 @@ def output_tau(
          This will be the hdfs or network version depending on settings.
         run_id (int): The current run id
         ultfoc_mapper (pd.DataFrame): The ULTFOC mapper DataFrame.
```
Comment (JenCheshire) [inline src/outputs/tau.py]: ultfoc mapper can be taken out too

**Review by JenCheshire — COMMENTED** [review]
Only small tidy up bits.  Code looks good, and all the outputs are looking as expected

**Review by JenCheshire — APPROVED** [review]
Fab

**Pull request merged**